### ISSUE TITLE
Mapping method matching not working for lower-bounded target type

### ISSUE THREAD

**Issue by ugrave** [mapstruct/mapstruct#775]
Example Objects to Map:

```
public class SourceObject {
        private List<String> numbers = new ArrayList<>();

        public List<String> getNumbers() {
            return numbers;
        }
 }
```

```
public class TargetObject {
        private List<Integer> numbers = new ArrayList<>();

        public TargetObject setNumbers(Iterable<? extends Integer> numbersToAdd) {
            for (Integer n : numbersToAdd) {
                numbers.add(n);
            }
            return this;
        }
}
```
The TargetObject is generated automatically by the [immutables](http://immutables.github.io/) framework.

The mapper looks like:

```
public abstract class TestMapper {

    public abstract TargetObject map(SourceObject source);

    public abstract List<Integer> mapToList(Collection<String> source);

    public Integer map(String value) {
        return Integer.valueOf(value);
    }
}
```

On mapper generation mapstructs generates an error on map method:
`Can't map property "java.util.List<java.lang.String> numbers" to "java.lang.Iterable<? extends java.lang.Integer> numbers". Consider to declare/implement a mapping method: "java.lang.Iterable<? extends java.lang.Integer> map(java.util.List<java.lang.String> value)".`

If i'm adding the following mapping method, everything is working:
```
public List<? extends Integer> mapToIterable(List<String> source) {
   return mapToList(source);
}
```
The generated mapper uses the mapToIterable to set the property on the target object. (This is my current solution i used for solve the problem)

I also try to change the return value ofthe mapToList method to `List<? extends Integer>`. But then i get the error:

```
Can't generate mapping method for a wildcard extends bound result
```

I have also try to define the mapping method to use with the IterableMapping annotation. But that does not help.

**Comment by agudian** [comment]
Fixed. Thanks for letting us know, Ulrich! :+1:

**Comment by amineRS2i** [comment]
Please can you tell us your solution !!!

**Comment by gunnarmorling** [comment]
@amineRS2i, this has been fixed in the 1.1.0.Beta1 release. Is this the version you are using?

**Comment by amineRS2i** [comment]
yes i i use this version, but i have open a ticket for my problem : 
[(https://github.com/mapstruct/mapstruct/issues/800)]
can you help me please ?
